corehq/apps/es/management/commands/resave_failed_forms_and_cases.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import (
+    print_function,
+    absolute_import
+)
+from six.moves import input
+from django.core.management.base import BaseCommand
+from corehq.apps.data_pipeline_audit.management.commands.compare_doc_ids import (
+    compare_xforms,
+    compare_cases,
+)
+from corehq.form_processor.interfaces.dbaccessors import (
+    FormAccessors,
+    CaseAccessors,
+)
+from corehq.form_processor.utils.xform import resave_form
+from corehq.apps.hqcase.utils import resave_case
+from corehq.util.log import with_progress_bar
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+        parser.add_argument('--xforms', action='store_true')
+        parser.add_argument('--cases', action='store_true')
+
+    def handle(self, domain, *args, **options):
+        resave_xforms = options.get('xforms')
+        resave_cases = options.get('cases')
+        if resave_xforms:
+            perform_resave_on_xforms(domain)
+        if resave_cases:
+            perform_resave_on_cases(domain)
+
+
+def perform_resave_on_xforms(domain):
+    xform_ids_missing_in_es, _ = compare_xforms(domain, 'XFormInstance')
+    print("%s Ids found for xforms missing in ES." % len(xform_ids_missing_in_es))
+    print(xform_ids_missing_in_es)
+    ok = input("Type 'ok' to continue: ")
+    if ok != "ok":
+        print("No changes made")
+        return
+    form_accessor = FormAccessors(domain)
+    for xform_id in with_progress_bar(xform_ids_missing_in_es, length=10):
```

Comment:
`length` should be `len(xform_ids_missing_in_es` or just left out entirely since `xform_ids_missing_in_es` is just a list

Reply:
Also you could use `for form in with_progress_bar(form_accessor.iter_forms(form_ids), length=len(form_ids))`

Reply:
Ah! wanted to modify the granularity/ number of cases per iteration.
Misinterpreted `granularity = min(50, length)` in `with_progress_bar`.
Thanks for catching. Will avoid passing it.